src/daemon/daemon.cpp

```diff
@@ -321,18 +321,28 @@ auto validate_create_arguments(const mp::LaunchRequest* request)
     auto option_errors = mp::LaunchError{};
 
     const auto opt_mem_size = try_mem_size(mem_size_str.empty() ? mp::default_memory_size : mem_size_str);
-    const auto opt_disk_space = try_mem_size(disk_space_str.empty() ? mp::default_disk_size : disk_space_str);
 
-    mp::MemorySize mem_size{}, disk_space{};
+    mp::MemorySize mem_size{};
     if (opt_mem_size && *opt_mem_size >= min_mem)
         mem_size = *opt_mem_size;
     else
         option_errors.add_error_codes(mp::LaunchError::INVALID_MEM_SIZE);
 
-    if (opt_disk_space && *opt_disk_space >= min_disk)
-        disk_space = *opt_disk_space;
-    else
-        option_errors.add_error_codes(mp::LaunchError::INVALID_DISK_SIZE);
+    // If the user did not specify a disk size, then 0 will be passed down. Otherwise, the specified size will be
+    // checked.
+    mp::MemorySize disk_space{}; // Zero by default.
```

Comment:
Didn't we say an `optional<mp::MemorySize>` would be cleaner than assigning a special meaning to "0"?

Reply:
Right, I only replaced it to pass down the command-line argument. Fixed.